### DISCUSSION TITLE
Add or update team repository permissions

### DISCUSSION THREAD

**Discussion by kiss90benedek** [community/community#50704] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm working on an automation that would add a team as 'maintainer' to a repo once they get added to `CODEOWNERS`.

I was going to call [this API](https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#add-or-update-team-repository-permissions) from GitHub Actions via `octokit/request-action`.

However, when I use a regular token, I get a 404. When I use a PAT, I get 403 saying PAT cannot be used here. How am I supposed to do this?

**Comment by Anmol-Baranwal** [comment, marked as answer]
It sounds like you are trying to use the GitHub API to automatically add a team as a maintainer to a repository when they are added to CODEOWNERS.

The reason you are receiving a 404 error when using a regular token is likely because the token does not have the necessary permissions to perform the action you are trying to carry out. To add a team as a maintainer to a repository, the token needs to have the "admin:org" scope.

On the other hand, the reason you are receiving a 403 error when using a PAT is likely because PATs cannot be used to authenticate for GraphQL API requests. In this case, you may need to use a personal access token (PAT) with the "admin:org" scope to authenticate for the GitHub API requests. I am not sure but you might need repo scope as well.

You can generate a new PAT with the necessary permissions by following these steps:

1. Go to your GitHub settings and select "Developer settings" -> "Personal access tokens".
2. Click on "Generate new token".
3. Give your token a name and select the necessary scopes, including "admin:org".
4. Click on "Generate token" and copy the generated token.

Once you have the PAT, you can use it to authenticate for the GitHub API requests in your automation script.

Here is an example script using the octokit/rest.js library and the generated PAT:

```javascript
const { Octokit } = require("@octokit/rest");
const octokit = new Octokit({
  auth: "YOUR_PAT_HERE"
});

// Replace the following variables with your own values
const teamSlug = "your-team-slug";
const owner = "your-repo-owner";
const repo = "your-repo-name";

octokit.graphql(`
  mutation {
    addTeamRepositoryPermission(
      teamId: "${teamSlug}",
      repositoryId: "${owner}/${repo}",
      permission: MAINTAIN
    ) {
      clientMutationId
    }
  }
`).then(response => {
  console.log("Team added as maintainer!");
}).catch(error => {
  console.error(error);
});
```
> Adjust the GraphQL query to match your specific use case, including the team ID, repository ID, and permission level

**Comment by kiss90benedek** [comment]
Thank you @Anmol-Baranwal. Originally, I tried to use a fine-grained access token, but a "admin:org" classic token worked.

(I used the API I posted originally. The docs had a typo: I had to use the arg as "permission" and not "permissions".)

**Reply by Anmol-Baranwal** [reply]
Thanks for letting me know.
